src/main/java/net/glowstone/command/minecraft/FillCommand.java

```diff
@@ -0,0 +1,249 @@
+package net.glowstone.command.minecraft;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import net.glowstone.block.GlowBlock;
+import net.glowstone.block.state.BlockStateData;
+import net.glowstone.block.state.InvalidBlockStateException;
+import net.glowstone.block.state.StateSerialization;
+import net.glowstone.command.CommandUtils;
+import net.glowstone.constants.ItemIds;
+import net.glowstone.i18n.LocalizedStringImpl;
+import org.bukkit.ChatColor;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.command.CommandSender;
+
+public class FillCommand extends GlowVanillaCommand {
+
+    /**
+     * Creates the instance for this command.
+     */
+    public FillCommand() {
+        super("fill");
+        setPermission("minecraft.command.fill"); // NON-NLS
+    }
+
+    @Override
+    public boolean execute(CommandSender sender, String label, String[] args,
+            CommandMessages commandMessages) {
+        if (!testPermission(sender, commandMessages.getPermissionMessage())) {
+            return true;
+        }
+        if (args.length < 7) {
+            sendUsageMessage(sender, commandMessages);
+            return false;
+        }
+        String itemName = CommandUtils.toNamespaced(args[6].toLowerCase());
+        Material type = ItemIds.getBlock(itemName);
+        if (type == null) {
+            new LocalizedStringImpl("setblock.invalid.type",
+                    commandMessages.getResourceBundle())
+                    .sendInColor(ChatColor.RED, sender, itemName);
+            return false;
+        }
+        byte dataValue = 0;
+        if (args.length > 7) {
+            String state = args[7];
+            BlockStateData data = CommandUtils.readState(sender, type, state);
+            if (data == null) {
+                return false;
+            }
+            if (data.isNumeric()) {
+                dataValue = data.getNumericValue();
+            } else {
+                try {
+                    dataValue = StateSerialization.parseData(type, data).getData();
+                } catch (InvalidBlockStateException e) {
+                    sender.sendMessage(ChatColor.RED + e.getMessage());
+                    return false;
+                }
+            }
+        }
+        Location location1 = CommandUtils.getLocation(sender, args[0], args[1], args[2]);
+        Location location2 = CommandUtils.getLocation(sender, args[3], args[4], args[5]);
+        double x1 = location1.getX();
+        double y1 = location1.getY();
+        double z1 = location1.getZ();
+        double x2 = location2.getX();
+        double y2 = location2.getY();
+        double z2 = location2.getZ();
+        double xmin = 0;
+        double ymin = 0;
+        double zmin = 0;
+        double xmax = 0;
+        double ymax = 0;
+        double zmax = 0;
+        double placeX;
+        double placeY;
+        double placeZ;
+        if (x1 > x2) {
```

Comment:
All of these can be replaced using `Math.max()` and `Math.min()`, used at the declaration of each variable.